File: python/Minimum_Pair_Removal_to_Sort_Array_I_3507.py
class Solution:
    def minimumPairRemoval(self, nums: List[int]) -> int:
        numsSize = len(nums)
        count = 0
        nonDecr = False
        while not nonDecr:
            minPair = 0;
            for i in range(numsSize - 1):
                if nums[i] + nums[i + 1] < nums[minPair] + nums[minPair + 1]:
                    minPair = i
            nonDecr = True
            for i in range(numsSize - 1):
                if nums[i] > nums[i + 1]:
                    nonDecr = False
                    count += 1
                    numsSize -= 1
                    nums[minPair] += nums[minPair + 1]
                    for j in range(minPair + 1, numsSize):
                        nums[j] = nums[j + 1]
                    break
        return count

File: cpp/Minimum_Pair_Removal_to_Sort_Array_I_3507.cpp
class Solution {
public:

    int minimumPairRemoval(vector<int>& nums) {
        int iter = 0;
        int numSize = nums.size();
        bool nonDecr = false;
        while (!nonDecr) {
            int minPair = 0;
            for (int i = 0; i < numSize - 1; i++) {
                if (nums[i] + nums[i + 1] < nums[minPair] + nums[minPair + 1]) minPair = i;
            }
            nonDecr = true;
            for (int i = 0; i < numSize - 1; i++) {
                if (nums[i] > nums[i + 1]) {
                    nonDecr = false;
                    iter += 1;
                    nums[minPair] += nums[minPair + 1];
                    for (int j = minPair + 1; j < numSize - 1; j++) {
                        nums[j] = nums[j + 1];
                    }
                    numSize--;
                    break;
                }
            }
        }
        return iter;
    }
};
